reader.js

```diff
@@ -11,159 +11,173 @@ let WSS;
+                response.writeHead(400);
+                response.end('Wrong ids provided' );
+                return;
+            }
 
-                        let transmissions = [];
-                        result.forEach(point => {
-                            let transmission = {};
-                            let names = point.groupRows.map((groupRow) => groupRow.name);
-                            point.forEach(measurement => {
-                                transmission[names.shift()] = measurement.value;
-                            });
-                            transmissions.push(transmission);
-                        });
+            let query = `select * from`;
+            names.forEach(name => {
+                query += ` ${name},`;
+            });
+            query = query.slice(0, -1);
+            query += ` where (mission = '${mission}') and (id = '${id}')`;
+            queries.push(query);
 
-                        response.setHeader("Access-Control-Allow-Origin", '*');
-                        response.end(JSON.stringify(transmissions));
+        });
 
-                    }).catch(function (err) {
-                        console.error(`Error querying data from InfluxDB! ${err.stack}`);
+        return influx.query(queries);
 
-                        if (WSS.readyState !== WebSocket.OPEN) {
-                            return;
-                        }
+    }).then(result => {
 
-                        WSS.send(id + ':error:' + e);
-                    });
-                }).catch(function (err) {
-                    console.error(`Error getting measurements! ${err.stack}`);
+        let transmissions = [];
+        result.forEach(point => {
+            let transmission = {};
+            let names = point.groupRows.map((groupRow) => groupRow.name);
+            point.forEach(measurement => {
+                transmission[names.shift()] = measurement.value;
+            });
+            transmissions.push(transmission);
+        });
 
-                    if (WSS.readyState !== WebSocket.OPEN) {
-                        return;
-                    }
+        response.end(JSON.stringify(transmissions));
 
-                    WSS.send(id + ':error:' + e);
-                });
-            }).catch(function (err) {
-                console.error(`Error connecting to InfluxDB! ${err.stack}`);
+    }).catch(function (err) {
+        console.error(`Error querying data from InfluxDB! ${err.stack}`);
 
-                if (WSS.readyState !== WebSocket.OPEN) {
-                    return;
-                }
+        response.writeHead(500);
+        response.end(JSON.stringify({error: id + ':error:' + e}));
 
-                WSS.send(id + ':error:' + e);
-            });
-        }
-    }
+    }).catch(function (err) {
+        console.error(`Error getting measurements! ${err.stack}`);
+
+        response.writeHead(500);
+        response.end(JSON.stringify({error: id + ':error:' + e}));
```

Comment:
Take a look at what you're writing as a response here -- maybe you want `response.end(JSON.stringify({error: err.message}));`?